source/val/validate_composites.cpp

```diff
@@ -376,6 +376,27 @@ spv_result_t ValidateCopyObject(ValidationState_t& _, const Instruction* inst) {
     return _.diag(SPV_ERROR_INVALID_DATA, inst)
            << "Expected Result Type and Operand type to be the same";
   }
+
+  const Instruction* type_inst = _.FindDef(result_type);
+  assert(type_inst);
+
+  const SpvOp type_opcode = type_inst->opcode();
+  if (type_opcode == SpvOpTypePointer &&
```

Comment:
OpCopyObject does not require variable pointers at all.

Reply:
What messed me up was the definition of variable pointer says that the pointer result of an OpCopyObject is a variable pointer.